src/librustc_privacy/lib.rs

```diff
@@ -1131,14 +1118,10 @@ impl<'a, 'tcx> SanePrivacyVisitor<'a, 'tcx> {
                     check_inherited(tcx, i.span, i.vis);
                 }
             }
-            hir::ItemEnum(ref def, _) => {
```

Comment:
Yeah this is fine, it's just checking that inside functions any non-inherited visibility [yield an error](https://play.rust-lang.org/?gist=a877d8d01e91cf4b21cb&version=stable) but if variants don't have visibility there's no need to check.